solidity/test/liquidation-tests/multiDepositBlackSwan.js

```diff
@@ -0,0 +1,158 @@
+      await testDeposit.retrieveSignerPubkey()
+
+      // provide the funding proof and move to ACTIVE state.
+      await testDeposit.provideBTCFundingProof(
+        _version,
+        _txInputVector,
+        _txOutputVector,
+        _txLocktime,
+        _fundingOutputIndex,
+        _merkleProof,
+        _txIndexInBlock,
+        _bitcoinHeaders,
+      )
+      deposits.push(testDeposit)
+    }
+  })
+
+  it("trade TDTs for TBTC", async () => {
+      for(let i = 0; i < deposits.length; i++) {
+        tdtId = await web3.utils.toBN(deposits[i].address)
+        await tbtcDepositToken.approve(vendingMachine.address, tdtId, {
+          from: depositInitiator,
+        })
+        await vendingMachine.tdtToTbtc(tdtId, {from: depositInitiator})
+      }
+    });
+
+  it("moves deposits to liquidation, large relative collateral value dip", async () => {
+    for(let i = 0; i < deposits.length; i++) {
+      await mockSatWeiPriceFeed.setPrice(new BN("4666666666666"))// price down 90%
+      await deposits[i].notifyUndercollateralizedLiquidation()
+
+      const depositState = await deposits[i].getCurrentState.call()
+      expect(depositState).to.eq.BN(states.LIQUIDATION_IN_PROGRESS)
+    }
+  })
+
+  it("Liquidate Deposits", async () => {
+    await createSnapshot()
+    let liquidatedIndex
+    let totalGasCost
+    const initialBalance = await web3.eth.getBalance(depositInitiator)
+    const gasPrice = await web3.eth.getGasPrice()
+
+    for(let i = 0; i < deposits.length; i++) {
+      const requirement = await deposits[i].getOwnerRedemptionTbtcRequirement.call(depositInitiator)
+      const totalRequirement = lotSizeTbtc.add(requirement)
+      const available = await tbtcToken.balanceOf(depositInitiator)
+      const duration = await tbtcConstants.getAuctionDuration.call()
+      await increaseTime(duration.toNumber())
+
+      if(available.gt(totalRequirement) || available.eq(totalRequirement)){
+        const tx0 = await tbtcToken.approve(deposits[i].address, totalRequirement, {from: depositInitiator})
+        const tx1 = await deposits[i].purchaseSignerBondsAtAuction({ from: depositInitiator })
+        const tx2 = await deposits[i].withdrawFunds({ from: depositInitiator })
+        const gasUSed = new BN(tx0.receipt.cumulativeGasUsed).add(new BN(tx1.receipt.cumulativeGasUsed).add(new BN(tx2.receipt.cumulativeGasUsed)))
+        const totalTxCost = new BN(gasPrice).mul(new BN(gasUSed))
+        totalGasCost = new BN(totalGasCost).add(totalTxCost)
+      }else{
+        liquidatedIndex = i
+        break
+      }
+    }
+    const expectedReward = new BN(minSufficientlyCollateralized).mul(new BN(liquidatedIndex))
+    const finalBalance = await web3.eth.getBalance(depositInitiator)
+    expect(liquidatedIndex,"Should be able to liquidate all but 1 deposit").to.eq.BN(deposits.length - 1)
+    expect(new BN(initialBalance).add(new BN(expectedReward))).to.eq.BN(new BN(finalBalance).add(totalGasCost))
```

Comment:
The variable names could be more descriptive out-of-context, eg. `minCollateralizedAmountETH`, but this test reads well and I understand the logic.